machine.go

```diff
@@ -459,17 +459,19 @@ func (m *Machine) startVMM(ctx context.Context) error {
 
 	errCh := make(chan error)
 	go func() {
-		if err := m.cmd.Wait(); err != nil {
-			m.logger.Warnf("firecracker exited: %s", err.Error())
+		waitErr := m.cmd.Wait()
+		if waitErr != nil {
+			m.logger.Warnf("firecracker exited: %s", waitErr.Error())
 		} else {
 			m.logger.Printf("firecracker exited: status=0")
 		}
 
-		if err := m.doCleanup(); err != nil {
-			m.logger.Errorf("failed to cleanup after VM exit: %v", err)
+		cleanupErr := m.doCleanup()
+		if cleanupErr != nil {
+			m.logger.Errorf("failed to cleanup after VM exit: %v", cleanupErr)
 		}
 
-		errCh <- err
+		errCh <- multierror.Append(waitErr, cleanupErr).ErrorOrNil()
```

Comment:
This change loses the error that was returned during the VM process. See line 431

Reply:
startVMM() returns the err and it is propagated to Start(). Do we want to receive the error from Wait()?

Reply:
The only difference I can see is that previously when `err` gets sets on line 498 (`err = m.waitForSocket`), that value would get picked up and put on `errCh` on line 474. However I think that behavior was somewhat buggy anyways in that if the machine starts and hangs without creating the socket file but also without the vmm process actually exiting, the call to `cmd.Wait` on line 462 could just hang indefinitely and, even if `cmd.Wait` did get unblocked, the goroutine on line 505 was never started and thus the error would never get set to `m.err`.

Given all that I think I am good with the change in this PR as it simplifies the code a bit, but it may be a good idea to also change line 501 to set `m.err = errors.New(msg)` instead of just `err`, which partially addresses the above issue by keeping the invariant that if there is an error during start, it will be set to `m.err` before `m.exitCh` is closed.

Reply:
Then maybe L444 as well? We can say that all errors in startVMM() must be accessible through exitCh

Reply:
Yeah good call, it should be set to `m.err` there too for consistency

Reply:
I think we should rename the err field or document that the error is only from the Wait call, if we are to do this. As it sits err seems like any error that was returned

Reply:
How about `startErr`? I agree that "err records any error executing the VMM" is too broad.

Reply:
Ah, but we are reusing the field for "any errors from Wait()"

Reply:
I think a reasonable definition for `m.err` would be "any error that occurs during or after machine start". We can write the doc for `Wait()` as "returns any error that occurs during or after machine start once the vmm process has exited". That seems to be what the intention of it was, right?

Reply:
I've renamed `m.err` to `m.fatalErr` on https://github.com/firecracker-microvm/firecracker-go-sdk/pull/153/commits/c32330d0013580fd07adc64145a7e4baeb28dfaf.

Reply:
I think I have addressed all comments on this thread. PTAL!